src/UnitTest/test_local_cache.cpp

```diff
@@ -1,215 +1,254 @@
+    local_cache_add(__FUNCTION__, now() + 60, sizeof(data), data);
+
+    assert(nullptr != local_cache_get(__FUNCTION__));
+    local_cache_clear();
+    assert(nullptr == local_cache_get(__FUNCTION__));
+
+    TEST_PASSED();
+    return true;
+}
+
+//
+// Add an item to the cache with a short expiry, then ensure it actually
+// expires.
+//
+static boolean VerifyExpiryWorks()
+{
+    TEST_START();
+
+    int expiry_seconds = 2;
+
+    // add an entry with expiry shortly in the future
+    static const uint8_t data[] = "stuff goes here";
+    local_cache_add(__FUNCTION__, now() + expiry_seconds, sizeof(data), data);
+
+    // ensure the data is there
+    assert(nullptr != local_cache_get(__FUNCTION__));
+
+    // wait for expiry, after which the data should be gone
+
+#if defined(__LINUX__)
+    sleep(expiry_seconds);
+#else
+    Sleep(expiry_seconds * 1000);
+#endif
+
+    assert(nullptr == local_cache_get(__FUNCTION__));
+
+    TEST_PASSED();
+
+    return true;
+}
+
+template <typename ExceptionT>
+static void AssertException(void (*function)())
+{
+    try
+    {
+        function();
+    }
+    catch (ExceptionT&)
+    {
+        return;
+    }
+
+    assert(!"Expected exception was not thrown");
+}
+
+//
+// Verify invalid parameter handling
+//
+static boolean InvalidParams()
+{
+    TEST_START();
+
+    static const uint8_t data[] = "test data";
+
+    AssertException<std::runtime_error>(
+        [] { local_cache_add("", now(), sizeof(data), data); });
+
+    AssertException<std::runtime_error>(
+        [] { local_cache_add(__FUNCTION__, now(), 0, data); });
+
+    AssertException<std::runtime_error>(
+        [] { local_cache_add(__FUNCTION__, now(), sizeof(data), nullptr); });
+
+    AssertException<std::runtime_error>([] { local_cache_get(""); });
+
+    TEST_PASSED();
+    return true;
+}
+
+//
+// Spawns multiple threads, intentionally creating a lot of contention for
+// a single cache entry.
+//
+static boolean ThreadSafetyTest()
+{
+    TEST_START();
+
+    // Pre-fill a data vector that's of sufficient size that we
+    // are likely to get conflicts between threads.
+    std::vector<uint8_t> data(64 * 1024);
+    for (size_t i = 0; i < data.size(); ++i)
+    {
+        data[i] = i & 0xff;
+    }
+
+    constexpr unsigned THREAD_LOOP_COUNT = 128;
+    const std::string ID = "data identifier";
+
+    auto cache_writer = [&](void) {
+        for (unsigned i = 0; i < THREAD_LOOP_COUNT; ++i)
+        {
+            local_cache_add(ID, now() + 60, data.size(), data.data());
+        }
+    };
+
+    auto cache_reader = [&](void) {
+        for (unsigned i = 0; i < THREAD_LOOP_COUNT; ++i)
+        {
+            auto retrieved = local_cache_get(ID);
+            assert(retrieved != nullptr);
+            assert(*retrieved == data);
+        }
+    };
+
+    // prime the cache entry first (in case a read thread runs first)
+    local_cache_add(ID, now() + 60, data.size(), data.data());
```

Comment:
Nit: Re-use of magic number 60. Suggestion to define it as a local and for readability.